controllers/fenceagentsremediation_controller.go

```diff
@@ -146,17 +152,36 @@ func (r *FenceAgentsRemediationReconciler) getFAPod(namespace string) (*corev1.P
 	}
 	if len(pods.Items) == 0 {
 		r.Log.Info("No Fence Agent pods were found")
-		podNotFoundErr := &errors.StatusError{ErrStatus: metav1.Status{
+		podNotFoundErr := &apiErrors.StatusError{ErrStatus: metav1.Status{
 			Status: metav1.StatusFailure,
 			Code:   http.StatusNotFound,
 			Reason: metav1.StatusReasonNotFound,
 		}}
 		return nil, podNotFoundErr
 	}
 	return &pods.Items[0], nil
+}
 
+// buildFenceAgentParams collects the FAR's parameters for the node based on the FARTemplate and FAR CRs
+func buildFenceAgentParams(farTemplate *v1alpha1.FenceAgentsRemediationTemplate, far *v1alpha1.FenceAgentsRemediation) ([]string, error) {
+	var fenceAgentParams []string
+	for paramName, paramVal := range farTemplate.Spec.SharedParameters {
+		fenceAgentParams = appendParamToSlice(fenceAgentParams, string(paramName), paramVal)
+	}
+
+	nodeName := v1alpha1.NodeName(far.Name)
+	for paramName, nodeMap := range farTemplate.Spec.NodeParameters {
+		if nodeMap[nodeName] != "" {
+			fenceAgentParams = appendParamToSlice(fenceAgentParams, string(paramName), nodeMap[nodeName])
```

Comment:
Nit:  I suggest keeping `paramName` type (if necessary convert is as late as possible).

Reply:
I don't understand your suggestion. May you elaborate please?

Reply:
Sure, here is a code snippet:

 
nodeName := v1alpha1.NodeName(far.Name)
for paramName, nodeMap := range farTemplate.Spec.NodeParameters {
	if nodeVal, isFound := nodeMap[nodeName]; isFound {
		fenceAgentParams = appendParamToSlice(fenceAgentParams, **paramName**, nodeVal)
...
 

 
func appendParamToSlice(fenceAgentParams []string, paramName **v1alpha1.ParameterName**, paramVal string) []string {
	if paramVal != "" {
		fenceAgentParams = append(fenceAgentParams, fmt.Sprintf("%s=%s", paramName, paramVal))
	} else {
		fenceAgentParams = append(fenceAgentParams, **string(paramName)**)
...

Reply:
Got it :)